sound/soc/sof/sof-client.c

```diff
@@ -265,6 +265,25 @@ int sof_client_ipc_tx_message(struct sof_client_dev *cdev, void *ipc_msg,
 }
 EXPORT_SYMBOL_NS_GPL(sof_client_ipc_tx_message, SND_SOC_SOF_CLIENT);
 
+int sof_client_ipc_set_get_data(struct sof_client_dev *cdev, void *ipc_msg,
+				bool set)
+{
+	if (cdev->sdev->pdata->ipc_type == SOF_IPC) {
+		struct sof_ipc_cmd_hdr *hdr = ipc_msg;
+
+		return sof_ipc_set_get_data(cdev->sdev->ipc, ipc_msg, hdr->size,
+					    set);
+	} else if (cdev->sdev->pdata->ipc_type == SOF_INTEL_IPC4) {
+		struct sof_ipc4_msg *msg = ipc_msg;
+
+		return sof_ipc_set_get_data(cdev->sdev->ipc, ipc_msg,
+					    msg->data_size, set);
+	}
+
+	return -EINVAL;
+}
+EXPORT_SYMBOL_NS_GPL(sof_client_ipc_set_get_data, SND_SOC_SOF_CLIENT);
```

Comment:
use "return ipc->ops->set_get_data(ipc->sdev, msg_data, msg_bytes, set); "
why we need to check ipc_type ? change it to ?
````c
int sof_client_ipc_set_get_data(struct sof_client_dev *cdev, void *ipc_msg, int size,
				bool set)
{
		return sof_ipc_set_get_data(cdev->sdev->ipc, ipc_msg, size, set);
}
````

Reply:
> use "return ipc->ops->set_get_data(ipc->sdev, msg_data, msg_bytes, set); " in sof_ipc_set_get_data

I just thought would be good practice to follow same implementation pattern as in sof_client_ipc_tx_message().

> why we need to check ipc_type ? change it to ?

The function prototype does not take message size separately and needs to cast the message pointer to a struct of the right form to take the size from message struct itself.